features/base/media/components/native/VideoTrack.js

```diff
@@ -0,0 +1,178 @@
+            fadeAnimation: new Animated.Value(1),
+            flipAnimation: new Animated.Value(1)
+        };
+    }
+
+    /**
+     * Renders video element with animation.
+     *
+     * @override
+     * @returns {ReactElement}
+     */
+    render() {
+        return (
+            <Animated.View
+                style = { [ styles.video, this._getAnimationStyles() ] }>
+                { super.render() }
+            </Animated.View>
+        );
+    }
+
+    /**
+     * Animates appearance of new video track.
+     *
+     * @private
+     * @param {Track} videoTrack - New video track to show.
+     * @returns {Promise}
+     */
+    _animateChangeVideoTrack(videoTrack) {
+        // If we're in process of animating, and new animation is going to
+        // start, stop the previous one first and reset animated values.
+        if (this._animation) {
+            this._animation.stop();
+            this.state.fadeAnimation.setValue(1);
+            this.state.flipAnimation.setValue(1);
+        }
+
+        // If we're switching between two local video tracks, that actually
+        // means we're switching local cameras, so we will use 'flip'-style
+        // animation in this case. In other case we will use 'fade'-style
+        // animation.
+        const animation = this.state.videoTrack.local && videoTrack.local
+            ? 'flipAnimation'
+            : 'fadeAnimation';
+
+        return this._animateHidePreviousVideoTrack(animation)
+            .then(() => this._animateShowNewVideoTrack(animation, videoTrack))
+            .catch(() => {
+                console.log('Animation was stopped');
+            });
+    }
+
+    /**
+     * Animates hiding of the current video.
+     *
+     * @param {string} animation - State variable to animate.
+     * @private
+     * @returns {Promise}
+     */
+    _animateHidePreviousVideoTrack(animation) {
+        return new Promise((resolve, reject) => {
+            this._animation = Animated.timing(
+                this.state[animation],
+                { toValue: 0 }
+            );
+
+            this._animation.start(result => {
+                this._animation = null;
+                result.finished ? resolve() : reject();
+            });
+        });
+    }
+
+    /**
+     * Animates showing of the new video.
+     *
+     * @param {string} animation - State variable to animate.
+     * @param {Track} videoTrack - New video track to show.
+     * @private
+     * @returns {Promise}
+     */
+    _animateShowNewVideoTrack(animation, videoTrack) {
+        return new Promise((resolve, reject) => {
+            super._changeVideoTrack(videoTrack);
+
+            this._animation = Animated.timing(
+                this.state[animation],
+                { toValue: 1 }
+            );
+
+            this._animation.start(result => {
+                this._animation = null;
+                result.finished ? resolve() : reject();
+            });
+        });
+    }
```

Comment:
The way I read _animateHidePreviousVideoTrack & _animateShowNewVideoTrack they are pretty much duplicates. (1) The toValue could be a function argument. (2) The super._changeVideoTrack could be moved outside of _animateShowNewVideoTrack. Anyway, I'll do the deduplication myself because I've already put sufficient effort in reviewing this PR but please note that duplication is forbidden by our coding style.